lib/qbo_api.rb

```diff
@@ -98,17 +98,26 @@ def reconnect
   end
 
   def request(method, path:, entity: nil, payload: nil, params: nil)
-    raw_response = connection.send(method) do |req|
-      path = finalize_path(path, method: method, params: params)
+    raw_response = raw_request(method, conn: connection, path: path, params: params, payload: payload)
+    response(raw_response, entity: entity)
+  end
+
+  def raw_request(method, conn:, path:, payload: nil, params: nil)
+    path = finalize_path(path, method: method, params: params)
+    @last_response = conn.public_send(method) do |req|
       case method
       when :get, :delete
         req.url path
       when :post, :put
         req.url path
         req.body = payload.to_json
+      else raise ArgumentError, "Unhandled request method '#{method.inspect}'"
       end
     end
-    response(raw_response, entity: entity)
+  end
+
+  def last_response
```

Comment:
What's the use for `#last_response`? Is it for Oauth2?

Reply:
@ntippie Sort of.  Basically, it's so that when you're handling an error in this gem, you're getting the body at most. But there's a lot more info in the last response object which you may want for various reasons, such as the quickbooks headers `Qbo-Version` and `Intuit-Tid` and maybe even `Content-Type`.  This just makes it accessible.  You think I should write a test using it or put in another PR.  This PR is really a stepping point for other work, so I could go either way.

Reply:
I can see it being useful for things like debugging, but I'm worried about keeping the entire response in memory unnecessarily. I've seen QBO API response bodies as large as 12MB. Maybe we can store those headers you mentioned as their own instance variables, like `@last_intuit_tid` and `@last_qbo_version`? Either way, maybe we should split this off into another PR so we can move this one forward.

What do you think, @minimul ?

Reply:
Let's leave off all instance vars (headers values included) this go-round.